src/util/pup_util.C

```diff
@@ -743,23 +744,23 @@ void PUP::toTextUtil::bytes(void *p,size_t n,size_t itemSize,dataType t) {
     for (size_t i=0;i<n;i++) {
       char *o=beginLine();
       switch(t) {
-      case Tshort: sprintf(o,"short=%d;\n",((short *)p)[i]); break;
-      case Tushort: sprintf(o,"ushort=%u;\n",((unsigned short *)p)[i]); break;
-      case Tint: sprintf(o,"int=%d;\n",((int *)p)[i]); break;
-      case Tuint: sprintf(o,"uint=%u;\n",((unsigned int *)p)[i]); break;
-      case Tlong: sprintf(o,"long=%ld;\n",((long *)p)[i]); break;
-      case Tulong: sprintf(o,"ulong=%lu;\n",((unsigned long *)p)[i]); break;
-      case Tfloat: sprintf(o,"float=%.7g;\n",((float *)p)[i]); break;
-      case Tdouble: sprintf(o,"double=%.15g;\n",((double *)p)[i]); break;
-      case Tbool: sprintf(o,"bool=%s;\n",((bool *)p)[i]?"true":"false"); break;
+      case Tshort: snprintf(o,maxCount-strlen(o),"short=%d;\n",((short *)p)[i]); break;
+      case Tushort: snprintf(o,maxCount-strlen(o),"ushort=%u;\n",((unsigned short *)p)[i]); break;
+      case Tint: snprintf(o,maxCount-strlen(o),"int=%d;\n",((int *)p)[i]); break;
+      case Tuint: snprintf(o,maxCount-strlen(o),"uint=%u;\n",((unsigned int *)p)[i]); break;
+      case Tlong: snprintf(o,maxCount-strlen(o),"long=%ld;\n",((long *)p)[i]); break;
+      case Tulong: snprintf(o,maxCount-strlen(o),"ulong=%lu;\n",((unsigned long *)p)[i]); break;
+      case Tfloat: snprintf(o,maxCount-strlen(o),"float=%.7g;\n",((float *)p)[i]); break;
+      case Tdouble: snprintf(o,maxCount-strlen(o),"double=%.15g;\n",((double *)p)[i]); break;
+      case Tbool: snprintf(o,maxCount-strlen(o),"bool=%s;\n",((bool *)p)[i]?"true":"false"); break;
 #if CMK_LONG_DOUBLE_DEFINED
-      case Tlongdouble: sprintf(o,"longdouble=%Lg;\n",((long double *)p)[i]);break;
+      case Tlongdouble: snprintf(o,maxCount-strlen(o),"longdouble=%Lg;\n",((long double *)p)[i]);break;
 #endif
 #ifdef CMK_PUP_LONG_LONG
-      case Tlonglong: sprintf(o,"longlong=%lld;\n",((CMK_PUP_LONG_LONG *)p)[i]);break;
-      case Tulonglong: sprintf(o,"ulonglong=%llu;\n",((unsigned CMK_PUP_LONG_LONG *)p)[i]);break;
+      case Tlonglong: snprintf(o,maxCount-strlen(o),"longlong=%lld;\n",((CMK_PUP_LONG_LONG *)p)[i]);break;
+      case Tulonglong: snprintf(o,maxCount-strlen(o),"ulonglong=%llu;\n",((unsigned CMK_PUP_LONG_LONG *)p)[i]);break;
 #endif
-      case Tpointer: sprintf(o,"pointer=%p;\n",((void **)p)[i]); break;
+      case Tpointer: snprintf(o,maxCount-strlen(o),"pointer=%p;\n",((void **)p)[i]); break;
```

Comment:
These lines don't need `-strlen(o)` since `o` is only printed to once and never advanced. Changing the variable definition `char * const o=beginLine();` could validate that.